splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceSystemProcedures.java

```diff
@@ -749,41 +750,43 @@ static public void createSysUtilProcedures(List<Procedure> procedures) {
                 .build();
         procedures.add(minRetentionPeriod);
 
-        Procedure snapshotSchema = Procedure.newBuilder().name("SNAPSHOT_SCHEMA")
+        Procedure snapshotSchema = Procedure.newBuilder().name("CREATE_SCHEMA_SNAPSHOT")
                 .varchar("schemaName", 128)
-                .varchar("snapshotName", 128)
+                .sqlControl(RoutineAliasInfo.MODIFIES_SQL_DATA)
                 .numOutputParams(0)
-                .numResultSets(0)
-                .ownerClass(SpliceAdmin.class.getCanonicalName())
+                .numResultSets(1)
+                .ownerClass(SnapshotSystemProcedure.class.getCanonicalName())
                 .build();
         procedures.add(snapshotSchema);
 
+        Procedure cloneSnapshot = Procedure.newBuilder().name("CLONE_SCHEMA_SNAPSHOT")
+                .bigint("snapshotId")
+                .catalog("schemaName")
+                .sqlControl(RoutineAliasInfo.MODIFIES_SQL_DATA)
+                .numOutputParams(0)
+                .numResultSets(1)
+                .ownerClass(SnapshotSystemProcedure.class.getCanonicalName())
+                .build();
+        procedures.add(cloneSnapshot);
+
         Procedure snapshotTable = Procedure.newBuilder().name("SNAPSHOT_TABLE")
                 .varchar("schemaName", 128)
                 .varchar("tableName", 128)
                 .varchar("snapshotName", 128)
                 .numOutputParams(0)
                 .numResultSets(0)
-                .ownerClass(SpliceAdmin.class.getCanonicalName())
+                .ownerClass(SnapshotSystemProcedure.class.getCanonicalName())
                 .build();
         procedures.add(snapshotTable);
 
         Procedure deleteSnapshot = Procedure.newBuilder().name("DELETE_SNAPSHOT")
-                .varchar("snapshotName", 128)
+                .bigint("snapshoId")
                 .numOutputParams(0)
-                .numResultSets(0)
-                .ownerClass(SpliceAdmin.class.getCanonicalName())
+                .numResultSets(1)
+                .ownerClass(SnapshotSystemProcedure.class.getCanonicalName())
```

Comment:
please put this into a function
`public static void addProcedures(List<Procedure> procedures) {`
in SnapshotSystemProcedure, see `TableSplit.addProcedures`.